src/molecules/service.py

```diff
@@ -182,17 +193,23 @@ def process_csv_file(self, file: UploadFile):
                     raise InvalidSmilesException(row["smiles"])
                 self.save(MoleculeRequest(smiles=row["smiles"], name=row["name"]))
                 number_of_molecules_added += 1
-            except Exception as e:
+            except InvalidSmilesException as e:
                 """
                 We are ignoring the exception and continuing to the next line. It is not important
                 whole file to be valid, we just want to add as many molecules as possible.
                 """
-                print(e)
+                logger.warning(
+                    f"Encountered invalid SMILES string: {e.smiles} in row: {row}"
+                )
+            except DuplicateSmilesException as e:
+                logger.warning(f"Duplicate SMILES string: {e.smiles} in row: {row}")
+            except Exception as e:
+                logger.error(f"Error processing row: {row}, error: {e}")
+
```

Comment:
Here is the only place so far where I had prints and replace with logging. In the future, I will try to expand my knowledge and use logging as much as possible.